### ISSUE TITLE
Ability to track status of optimize api

### ISSUE THREAD

**Issue by ppf2** [elastic/elasticsearch#10008]
Optimize is an intensive process and can run for many hours.  It will be nice to provide a way to track its status (esp. when we switch to a sync operation in the future: https://github.com/elasticsearch/elasticsearch/pull/9640).   

This is likely dependent on task management:
https://github.com/elasticsearch/elasticsearch/issues/6914

**Comment by kjelle** [comment]
+1

E.g. im optimizing a 24shard index from 1127 segments down to max_num_segments=1... it would be nice to see that it is actually moving forward, estimate how long it will take etc.

**Comment by clintongormley** [comment]
Closing in favour of https://github.com/elastic/elasticsearch/issues/15117
